src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -22,10 +22,23 @@
 #include <ripple/app/main/Application.h>
 #include <ripple/app/misc/Validations.h>
 #include <ripple/app/data/DatabaseCon.h>
+#include <ripple/core/ConfigSections.h>
 #include <boost/format.hpp>
+#include <boost/tokenizer.hpp>
 
 namespace ripple {
 
+namespace detail
+{
+/** Amendments that are "compiled into" this app
+
+    Add amendments to this collection at build time to enable them on this
+   server. The first element of the pair is the id, the second element is
+   the "friendly name".
+*/
+std::set<std::pair<std::string, std::string>> PreEnabledAmendments;
```

Comment:
The reason preEnabledAmendments is a pair of string is because there is no way to tell if an invalid hex string was fed to the uint64(string) constructor. So if I change preEnabledAmendments to take a pair of <uint256,string> and initialize it with {uint256("123XXX"), "invalid_first"}, then the program will not spot the error. As written, it will throw on this error. The use-case for preEnabledAmendments is human-entered hashes, so throwing on an easily made typo is important.

I agree with your point about pairs of string,string easy to get wrong. This was the reason I changed the sig of addKnown. I could change the implementation of base_uint to try to handle this, but I'm *really* hesitant to much around with such a basic type. I think the string,string pair is actually less error prone for this use case. I think <string,string> pair is better for this case.